### ISSUE TITLE
0 bpm/frozen scanline support?

### ISSUE THREAD

**Issue by Launfer** [Cytoid/Cytoid#104]
Will there be support for frozen scanlines in the future? And if so, when might it become a thing? I'm not sure if anything has been posted already regarding this, but I've read around Cytoid wiki/discord a bit and couldn't find anything about it so I thought it would be a good idea to just pose the question here. Even though PCtyx supports it, it converts to a sweeping scanline when I compile a chart into Cytoid, so I have been just assuming it's not supported because I can't get it to work despite my best efforts. If by some chance it IS supported and I'm just being dumb please tell me if there's any wacky stuff I need to do to the chart file to make the frozen scanline work in Cytoid. Thanks a bunch!

**Comment by Weathercold** [comment]
I’m no charter but here’s a quote from jcexe:
```
-Speed must be bigger than 0 and thus cannot be truly stopped in any way. 
-Direction cannot be changed in the middle of its scrolling.

afaik there are workarounds/strange properties c2 scanline can do though.
- If the start tick of the scanline is in negative number, the scanline "froze" at the opposite position. (how pctyx's handle freeze scanline when exported as c2 text file)
- If direction is 0, the scanline "freeze" in the exact middle of the screen. (doesn't look pretty especially if your scanline in pctyx isn't exactly at the middle and have very little niche uses)
- If the scanline is absent between ticks (delete the lines in the file), the scanline scrolls out of bounds.
```

**Comment by Teages** [comment]
The frozen scanline is supported.

Before we talk about supporting frozen scanline. Let us review how to stop the line.

Cytoid
------------------------
Here is the way of Cytoid & C2pPlayer. (in fact, they are same)

> turn the strat tick of the page to -2147483648

This way is easy but what it can do is limited. We can only stop the line on the top/bottom. 

Cytus II
------------------------
In Cytus 2, Charters in Rayark use a understandable and more applicable to a wider range way.

1. Turn the `tempo` smaller. (Means BPM bigger, BPM = 60,000,000 / tempo) (is unnecessary but can improve accuracy);
2. Turn the 'tempo' to the biggest value (60,000,000) (Means the smallest BPM, 1/60,000,000 but not 0) for a while, and then
3. Turn the 'tempo' back to normal. 

In this way, you can stop the line everywhere you want.

With it, now you know why BPM can not be 0, because tempo can not be gigantic.(60,000,000 MAX)

If you are using PCTyx, just load the story which saved with your chart file (maybe ***-sb.json).

**Comment by Launfer** [comment]
> The frozen scanline is supported.
> 
> Before we talk about supporting frozen scanline. Let us review how to stop the scanline.
> 
> ## Cytoid
> Here is the way of Cytoid & C2pPlayer. (in fact, they are same)
> 
> > turn the `start_tick` of the page to -2147483648
> 
> This way is easy but what it can do is limited. We can only stop the line on the top/bottom.
> 
> ## Cytus II
> In Cytus 2, Charters in Rayark use a understandable and more applicable to a wider range way.
> 
> 1. Turn the `tempo` smaller. (Means BPM bigger, BPM = 60,000,000 / tempo) (is unnecessary but can improve accuracy);
> 2. Turn the `tempo` to the biggest value (60,000,000) (Means the smallest BPM, 1/60,000,000 but not 0) for a while, and then
> 3. Turn the `tempo` back to normal.
> 
> In this way, you can stop the line everywhere you want.
> 
> With it, now you know why BPM can not be 0, because tempo can not be gigantic.(60,000,000 MAX)
> 
> If you are using PCTyx, just load the storyboard which saved with your chart file (maybe ***-sb.json).



> The frozen scanline is supported.
> 
> Before we talk about supporting frozen scanline. Let us review how to stop the scanline.
> 
> ## Cytoid
> Here is the way of Cytoid & C2pPlayer. (in fact, they are same)
> 
> > turn the `start_tick` of the page to -2147483648
> 
> This way is easy but what it can do is limited. We can only stop the line on the top/bottom.
> 
> ## Cytus II
> In Cytus 2, Charters in Rayark use a understandable and more applicable to a wider range way.
> 
> 1. Turn the `tempo` smaller. (Means BPM bigger, BPM = 60,000,000 / tempo) (is unnecessary but can improve accuracy);
> 2. Turn the `tempo` to the biggest value (60,000,000) (Means the smallest BPM, 1/60,000,000 but not 0) for a while, and then
> 3. Turn the `tempo` back to normal.
> 
> In this way, you can stop the line everywhere you want.
> 
> With it, now you know why BPM can not be 0, because tempo can not be gigantic.(60,000,000 MAX)
> 
> If you are using PCTyx, just load the storyboard which saved with your chart file (maybe ***-sb.json).

Yeah I probably should have specified that I did manage to get the scanline to stop (in the middle of the play area as well), but that it was behaving differently to how it was working in PCtyx. My bad. But from my own experience and what's been said, I gather it IS possible (to get it to work how I want), but requires some more technical fiddling and is still limited in its application. I guess I actually meant support in a way that would be easily translated when I take a PCtyx chart and turn it into a Cytoid chart.
